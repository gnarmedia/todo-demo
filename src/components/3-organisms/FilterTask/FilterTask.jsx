// import dependencies
import React from "react";
import PropTypes from "prop-types";

/**
 * @name FilterTask
 * @desc Filters for tasks
 * @param {string} filter
 * @returns {function}
 */
const FilterTask = ({ onFilterClick }) => {
    const filters = ["All", "Incomplete", "Complete"];

    return (
        <ul>
            {filters.map(filter => {
                return (
                    <button
                        key={`filterTask_${filter}`}
                        onClick={() => onFilterClick(filter.toUpperCase())}
                    >
                        {filter}
                    </button>
                );
            })}
        </ul>
    );
};

FilterTask.propTypes = {
    onFilterClick: PropTypes.func.isRequired
};

// export our component for importing into other modules
export default FilterTask;
