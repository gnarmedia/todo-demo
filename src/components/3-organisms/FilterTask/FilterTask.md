```jsx
function handleFilterClick(filter) {
  const util = require('util');
  alert(`set filter: ${filter}`);
  return false;
};
<FilterTask onFilterClick={handleFilterClick} />
```