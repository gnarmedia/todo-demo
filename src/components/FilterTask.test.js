import React from "react";
import { shallow } from "enzyme";

// import components
import FilterTask from "./FilterTask.jsx";

describe("FilterTask", () => {
    it("should render the FilterTask", () => {
        const handleFilterClickMock = jest.fn(),
            wrapper = shallow(
                <FilterTask onFilterClick={handleFilterClickMock} />
            );

        expect(wrapper).toMatchSnapshot();
    });

    // add event handler test
});
