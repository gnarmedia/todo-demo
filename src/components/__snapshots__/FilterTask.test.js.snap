// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterTask should render the FilterTask 1`] = `
<ul>
  <button
    key="filterTask_All"
    onClick={[Function]}
  >
    All
  </button>
  <button
    key="filterTask_Incomplete"
    onClick={[Function]}
  >
    Incomplete
  </button>
  <button
    key="filterTask_Complete"
    onClick={[Function]}
  >
    Complete
  </button>
</ul>
`;
